index.js

```diff
@@ -1,118 +1,171 @@
-    [' ',' ',' ',' ',' ',' ',' ',' '],
-    [' ',' ',' ','w','b',' ',' ',' '],
-    [' ',' ',' ','b','w',' ',' ',' '],
-    [' ',' ',' ',' ',' ',' ',' ',' '],
-    [' ',' ',' ',' ',' ',' ',' ',' '],
-    [' ',' ',' ',' ',' ',' ',' ',' '],    
-  ];
-
-  drawBoard(board);
-
-  //for each row of the board
-    //for each space in this row
-      //if a user clicks on this space
-        //if this space is empty
-          //if this is player black
-
-          //if this is player white
-        //else
-          //alert the user that this space is occupied
+  //if (1)southwest([row + 1][col - 1]) adjacent is opposite color
+  if(board[row + 1][col - 1] === opponent) {
+    vertical = 1;
+    horizontal = -1;
+    const flipped = piecesToFlip(row + 1, col - 1, vertical, horizontal, player);
+    if(flipped > 0) {
+      piecesFlipped = true;
+    }
+  }    
 
-}
+  //if (2)south([row + 1][col]) adjacent is opposite color
+  if(board[row + 1][col] === opponent) {
+    vertical = 1;
+    horizontal = 0;
+    const flipped = piecesToFlip(row + 1, col, vertical, horizontal, player);
+    if(flipped > 0) {
+      piecesFlipped = true;
+    }
+  }     
 
-//flip over pieces depending on the board state
-const updateState = () => {
+  //if (3)southeast([row + 1][col + 1]) adjacent is opposite color
+  if(board[row + 1][col + 1] === opponent) {
+    vertical = 1;
+    horizontal = 1;
+    const flipped = piecesToFlip(row + 1, col + 1, vertical, horizontal, player);
+    if(flipped > 0) {
+      piecesFlipped = true;
+    }
+  }    
+
+  //if (4)east([row][col + 1]) adjacent is opposite color
+  if(board[row][col + 1] === opponent) {
+    vertical = 0;
+    horizontal = 1;
+    const flipped = piecesToFlip(row, col + 1, vertical, horizontal, player);
+    if(flipped > 0) {
+      piecesFlipped = true;
+    }
+  }     
+  //if (5)northeast([row - 1][col + 1]) adjacent is opposite color
+  if(board[row - 1][col] === opponent) {
+    vertical = -1;
+    horizontal = 1;
+    const flipped = piecesToFlip(row - 1, col + 1, vertical, horizontal, player);
+    if(flipped > 0) {
+      piecesFlipped = true;
+    }
+  }     
+
+  //if (6)north([row - 1][col]) adjacent is opposite color
+  if(board[row][col-1] === opponent) {
+    vertical = -1;
+    horizontal = 0;
+    const flipped = piecesToFlip(row - 1, col, vertical, horizontal, player);
+    if(flipped > 0) {
+      piecesFlipped = true;
+    }
+  }     
+
+  //if (7)northwest([row - 1][col - 1]) adjacent is opposite color
+  if(board[row - 1][col - 1] === opponent) {
+    vertical = -1;
+    horizontal = -1;
+    const flipped = piecesToFlip(row - 1, col - 1, vertical, horizontal, player);
+    if(flipped > 0) {
+      piecesFlipped = true;
+    }
+  }     
+
+  //if any pieces were flipped
+  if(piecesFlipped) {
+    //this move is valid and the piece can be placed where the player has chosen
+    board[row][col] = player;
+    //change players
+    currentPlayer = opponent;

```

Comment:
You technically don't even need to store the value of the opponent:

```suggestion
    currentPlayer = !currentPlayer;
```

Reply:
I don't really understand that, can you tell me why?

Reply:
Because you're storing the `currentPlayer` as a boolean now, so all you have to do is flip the bit.

I guess this is one of those advantages of using `0`/`1` instead of strings.

Reply:
And you don't need to store the value of the opponent since computers are so good at binary manipulation that you could easily just use `!currentPlayer` anywhere you would use `opponent` (though it's fine to keep `opponent` if you think it's more clear).

Reply:
Derp, my bad. `currentPlayer`/`opponent` are _booleans_ which aren't technically the same thing as binary. So if `currentPlayer === 0` then `currentPlayer = !currentPlayer;` is really saying `currentPlayer = !0;` and `!0` is `true` not `1` since the esclamation point coerces to a boolean, then does the "not" 🤦 

You could technically use the [unary + operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Unary_plus) to convert the boolean back to binary like: `currentPlayer = +!currentPlayer;` but since that's a pretty obscure operator (I had to look it up), using the ternary operator may be more maintainable. Totally up to you!